community/kernel/src/main/java/org/neo4j/kernel/impl/store/StoreFactory.java

```diff
@@ -148,12 +148,6 @@ public NeoStore newNeoStore( boolean allowCreateEmpty )
             return createNeoStore();
         }
 
-        CountsTracker countsTracker = null;
-        if ( storeExists && countsStoreExists() )
-        {
-            countsTracker = newCountsStore();
-        }
-
```

Comment:
Yes. That's what needs fixing. My initial thought was to call `init()` in `rebuildCountStoreIfNeeded()`, but that didn't work in the other recovery scenarios. Now I'm thinking that another phase should be added. Possibly by using `start()`, or it could be better to move away from `Lifecycle` and use explicit names for these phases. My thinking is that `init()` would open the store if it exists, but go into a "pending" state if it doesn't, and then in that pending state, `start()` would create the file.